compiler/src/iree/compiler/Dialect/Stream/Analysis/Partitioning/ReferencePartitioning.cpp

```diff
@@ -386,6 +386,49 @@ partitionRegionConcurrencyReference(IREE::Stream::PartitioningConfigAttr config,
       // Always inherit hazards whether merging or not.
       opInfo.hazards |= userInfo.hazards;
     }
+
+    // Additional exhaustive testing for users of tied operands.
+    // For each resource operand of this op we scan back to see if there are any
```

Comment:
Not really "scan back", more like "scan forward"? When I read "back" I imagine that we'll traverse all users defined before the `op`

Reply:
ah yeah, waves are created in reverse, so it's back into the waves that have been defined - will clarify